Demo/Demo/Extensions/CustomButton.swift

```diff
@@ -0,0 +1,21 @@
+import UIKit
+
+class CustomButton: UIButton {
+    convenience init(title: String) {
+        self.init(type: .system)
+
+        setTitle(title, for: .normal)
+        translatesAutoresizingMaskIntoConstraints = false
+        titleLabel?.font = .preferredFont(forTextStyle: .body)
+        titleLabel?.numberOfLines = 0
+        titleLabel?.textAlignment = .center
```

Comment:
Can this just be an extension on `UIButton` instead of a new class?

Reply:
Extensions should be used carefully. I think it's easy for them to become a catch-all. I think a question to ask before making an extension is "Does the entire class need to know". So the entire UIButton class doesn't really need to know about our custom settings, so in that case a subclass makes more sense.

Reply:
My only hesitancy with making this an extension is if we ever want to have a button or text field that doesn't conform to these things. Now we have the flexibility to use custom buttons / text fields that looks the same while still being able to create a standard button or text field if needed. These were initially functions on an extension of UIViewController but were moved to be a subclass based on @scannillo suggestion. Sammy curious to hear your thoughts!

Reply:
Imo it's just shorter, no subclass needed, and if we don't use the new `init` in the extension then it's just gonna be a normal usual UIButton, so no harm risked :smile:
but it's just a minor suggestion, not a blocker, I'll approve once the other comments are addressed !!!! 🚀

Reply:
We shouldn't use an extension just because it is shorter. Extending UIViewController to expose custom buttons and views doesn't make much sense to me in terms of organization of logic. How often do you access buttons or UI elements via functions on UIViewController? Does that make sense? It doesn't feel like a native pattern. What if we want this custom button on a `UIView`, not on `ViewController`?

Also, it is common practice to create custom classes for custom views / what you see in tutorials often.

This is a demo app, so the stakes are super low. But, just in general thought it might help to explain where I am coming from.

We can now instantiate a `CustomButton` type in the same way we would instantiate a regular `UIButton`.

Reply:
ah what I meant was this:
```
extension UIButton {
    convenience init(title: String) {
        self.init(type: .system)
        ...
    }
}
```